unsupported/test/cxx11_tensor_reduction_hip.cu

```diff
@@ -0,0 +1,159 @@
+// This file is part of Eigen, a lightweight C++ template library
+// for linear algebra.
+//
+// Copyright (C) 2015 Benoit Steiner <benoit.steiner.goog@gmail.com>
+//
+// This Source Code Form is subject to the terms of the Mozilla
+// Public License v. 2.0. If a copy of the MPL was not distributed
+// with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#define EIGEN_TEST_NO_LONGDOUBLE
+#define EIGEN_TEST_NO_COMPLEX
+#define EIGEN_TEST_FUNC cxx11_tensor_reduction_hip
+#define EIGEN_USE_GPU
+
+#ifdef __NVCC__
+#if defined __CUDACC_VER__ && __CUDACC_VER__ >= 70500
+#include <cuda_fp16.h>
```

Comment:
probably need to remove this?

also in my other comment i'm wondering should we keep "_hip.cu" or just edit "_cuda.cu? in-place?

Reply:
/cc @dagamayank for awareness.

I have no particular preference on this topic. I think we should start submitting changes here upstream ASAP to hear from maintainers.

Reply:
My preference is to keep the backends separate for the reasons @deven-amd mentioned. We are having a real hard time with PyTorch with its of unifying the source. But eventually this decision is of the maintainer of Eigen.

Reply:
Filed issue #1552 (http://eigen.tuxfamily.org/bz/show_bug.cgi?id=1552) to get the ball rolling on submitting a PR to the Eigen codebase. Hoping someone will respond within a day or two.